src/com/epam/javatrainings/annotation/DTOValidatorImpl.java

```diff
@@ -0,0 +1,155 @@
+            if (field.isAnnotationPresent(Min.class)){
+                field.setAccessible(true);
+                minProcessor(field,t);
+            }
+            if (field.isAnnotationPresent(Max.class)){
+                field.setAccessible(true);
+                maxProcessor(field,t);
+            }
+        }
+
+        System.out.println(message);
+    }
+
+    private <T> void maxProcessor(Field field, T t) {
+        try{
+            Object value = field.get(t);
+            Max max = (Max) field.getAnnotation(Max.class);
+            int maxLimit = max.value();
+            if (value instanceof Integer){
+                maxLimitation((int)value,maxLimit);
+            }
+
+        } catch (IllegalAccessException e) {
+            e.printStackTrace();
+        }
+    }
+
+    private void maxLimitation(int value, int maxLimit) {
+        if (value > maxLimit){
+            message += "\nThe " + value + " is greater than max limit(" + maxLimit + ")";
+
+        }
+    }
+
+    private <T> void minProcessor(Field field, T t) {
+        try {
+            Object value = field.get(t);
+            Min min = (Min) field.getAnnotation(Min.class);
+            int minLimit = min.value();
+            if (value instanceof Integer){
+                minLimitation((int) value,minLimit);
+            }
+        } catch (IllegalAccessException e) {
+            e.printStackTrace();
+        }
+    }
+
+    private void minLimitation(int value,int minLimit) {
+        if (value < minLimit){
+            message += "\nThe " + value + " is less than min limit(" + minLimit + ")";
+        }
+    }
+
+    private <T> void adulthoodProcessor(Field field, T t) {
+        try {
+            Object value = field.get(t);
+            Adulthood adulthood = (Adulthood) field.getAnnotation(Adulthood.class);
+            if (value instanceof LocalDate){
+                validateAdulthood((LocalDate) value);
+            }
+        } catch (IllegalAccessException e) {
+            e.printStackTrace();
+        }
+    }
+
+    private void validateAdulthood(LocalDate value) {
+        if (LocalDate.now().compareTo(value) >= 18){
+            message += "\nIs adult";
+        }else{
+            message += "\nIsn't adult";
+        }
+    }
+
+    private <T> void emailProcessor(Field field, T t) {
+        try {
+            Object value = field.get(t);
+            Email email = (Email) field.getAnnotation(Length.class);
+            if (value instanceof String){
+                validateEmail((String) value);
+            }
+        } catch (IllegalAccessException e) {
+            e.printStackTrace();
+        }
+
+    }
+
+    private void validateEmail(String value) {
+        Matcher matcher = VALID_EMAIL_ADDRESS_REGEX.matcher(value);
+        if (matcher.find()){
+            message += "\nThe email is valid";
+        }else {
+            message += "\nThe email is invalid";
+        }
+    }
+
+    private <T> void lengthProcessor(Field field,T t){
```

Comment:
processors can be in separate classes, and maybe return a class Error {fieldname, errorMessage}